core/trino-spi/src/main/java/io/trino/spi/block/AbstractSingleRowBlock.java

```diff
@@ -21,7 +21,7 @@
 public abstract class AbstractSingleRowBlock
         implements Block
 {
-    protected final int rowIndex;
```

Comment:
you should add `protected abstract getRowIndex()` method instead of making `AbstractSingleRowBlock` a mutable class

Reply:
Oh, you right. It is better approach +1